Client/Frontend/Browser/PlaylistHelper.swift

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import WebKit
+import AVKit
+import Data
+import BraveShared
+import Shared
+
+private let log = Logger.browserLogger
+
+enum PlaylistItemAddedState {
+    case added
+    case pendingUserAction
+    case existing
+}
+
+protocol PlaylistHelperDelegate: NSObject {
+    func present(controller: UIViewController)
+    func showPlaylistToast(info: PlaylistInfo, itemState: PlaylistItemAddedState)
+    func addToPlayListActivity(info: PlaylistInfo?, itemDetected: Bool)
+}
+
+class PlaylistHelper: TabContentScript {
+    fileprivate weak var tab: Tab?
+    public weak var delegate: PlaylistHelperDelegate?
+    private var url: URL?
+    private var playlistItems = Set<String>()
+    private var urlObserver: NSObjectProtocol?
+    
+    init(tab: Tab) {
+        self.tab = tab
+        self.url = tab.url
+        
+        urlObserver = tab.webView?.observe(\.url, options: [.old, .new], changeHandler: { [weak self] _, change in
+            guard let self = self, let url = change.newValue else { return }
+            if self.url != url {
+                self.url = url
+                self.playlistItems = Set<String>()
+            }
+        })
+    }
+    
+    static func name() -> String {
+        return "PlaylistHelper"
+    }
+    
+    func scriptMessageHandlerName() -> String? {
+        return "playlistHelper"
+    }
+    
+    func userContentController(_ userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        
+        guard let item = PlaylistInfo.from(message: message),
+              !item.src.isEmpty else {
+            delegate?.addToPlayListActivity(info: nil, itemDetected: false)
+            return
+        }
+        
+        if item.duration <= 0.0 && !item.detected || item.src.isEmpty || item.src.hasPrefix("data:") {
```

Comment:
`item.src.hasPrefix("data:")`, should we check for both `data:` and `Data:`?

Reply:
No because `Data:` isn't a value URI scheme according to spec.